### ISSUE TITLE
Consider adding run selected text in terminal to editor context menu

### ISSUE THREAD

**Issue by Tyriar** [Microsoft/vscode#8128]
#7034/#8126 adds the new command `workbench.action.terminal.runSelectedText` to run the editor selection in the terminal. I think this would make sense in the editor's context menu.

**Comment by alexandrudima** [comment]
fyi @isidorn @jrieken @egamma 

I think it makes sense for some workflows and not for others. Same as `Debug: Add to watch` which also makes sense for some workflows and not for others. I don't know how that one went in, I was personally not consulted :).

I do not want to be the gate-keeper of the context menu and I much rather suggest that we invest on the context menu customization story by the end user.

I would personally like to remove `Debug: Add to watch` from **my** context menus, and add three times in each group the `Copy` (as thats the only one I use from time to time ;) ), etc. I think we should continue investing in the menu story that @jrieken started and come up with a user customization story.

**Comment by jrieken** [comment]
Yes - user customisable context menus

**Comment by isidorn** [comment]
@alexandrudima we only add the debug: add to watch if there is a launch.json present in the workspace of the user. I agree that still it sometimes does not make sense that it is present.
I am more for removing all of these than adding more - so I can remove debug: add to watch if the presence of a launch.json is not a clear enough indication that a user would find this useful

**Comment by jrieken** [comment]
@Tyriar I'll ping once the menu/context menu customisation is done

**Comment by alexandrudima** [comment]
@isidorn I'm not complaining, I'm exactly saying that I don't wanna be the gatekeeper here -- I cannot represent in my opinions all the users of the contextmenu :).

I mentioned the Debug one because it is similar in nature with the proposed run selected text in terminal, some folks might use it a lot, some folks might never use it. 

Anyways, I was trying to make the case for prioritizing end-user customization.

**Comment by Tyriar** [comment]
@alexandrudima thanks for the feedback, that's why I assigned it to you instead of going ahead and doing it :smiley: I'm also a fan of keeping the options slimmed down as much as possible for my workflow.

By user customization do you mean turning this on via a setting?

**Comment by alexandrudima** [comment]
@Tyriar No. @jrieken and I discussed last week how it would be very nice to expose each menu (editor contextmenu, editor primary actions -- the ones at the top, explorer contextmenu) to end users for configuration, such that an end user could remove a menu item, add a menu item, reorder menu items, etc.

**Comment by Tyriar** [comment]
@jrieken is there an issue for the context menu customization story?

**Comment by Tyriar** [comment]
@jrieken @alexandrudima It's still not clear to me, by configuration do you mean via settings.json (using something yet to be implemented), or via an extension using the new API in 1.3.0?

**Comment by jrieken** [comment]
It will be a new story, analog to `keybindings.json`. Think of a file `menus.json` that declared the menu structure you want.
